### PR TITLE
FIX: resolves issue with restrict_forward_to_label

### PR DESCRIPTION
**Pull request by clamus**
This resolves issue #2971.

### CODE REVIEW

**Comment by clamus** [comment]
Oops. I realized there is an additional issue with `restrict_forward_to_label`. After fixing `vertno` in `src`s for each hemisphere to actually map to vertex numbers in source space, now the ordering of of the columns of `fwd` solution do not match the ordering of the source space vertices. This definitely need to be fixed.

**Comment by Eric89GXL** [comment]
> the ordering of of the columns of fwd solution do not match the ordering of the source space vertices. This definitely need to be fixed.

Yeah that will need to be fixed. Let me know when I should look.

**Comment by clamus** [comment]
This should be ready for review now.

File: mne/forward/forward.py [hunk 1]
```diff
@@ -1297,16 +1298,28 @@ def restrict_forward_to_label(fwd, labels):
...
+    message = 'labels needs to be an instance of Label or a list of Label'
+    vertices = [np.array([], int), np.array([], int)]
 
     if not isinstance(labels, list):
         labels = [labels]
 
+    # Get vertices separately of each hemisphere from all label
+    for label in labels:
+        if not isinstance(label, Label):
+            raise TypeError(message)
```
Comment (agramfort) [inline mne/forward/forward.py]: add to the error message the type you got as input

**Comment by agramfort** [comment]
besides

LGTM

thanks heaps @clamus !

**Comment by clamus** [comment]
Added the input type to error message. Thanks @agramfort!

**Comment by agramfort** [comment]
Thx heaps

**Comment by clamus** [comment]
Closes #2971.

**Pull request merged**